### ISSUE TITLE
Add global config

### ISSUE THREAD

**Issue by tsimonq2** [tsujan/FeatherPad#93]
Lubuntu Next ships with FeatherPad as default (I've been using it for a bit now, it seems stable enough to replace JuffEd), and just in case others want to do this as well, it should be apparent for distributors on how to install a custom config should they need to do that.

**Comment by tsujan** [comment]
They could customize FeatherPad and find the customized config as `~/.config/featherpad/fp.conf`. It's better to remove the `[sessions]` part though.

**Comment by tsimonq2** [comment]
Right, but is there code in FeatherPad that, if that file doesn't exist under `.config`, it will read `$XDG_CONFIG_DIRS` and if a featherpad directory is in none of those directories, it will use the default one (and take the default config for things not explicitly defined)?

**Comment by tsujan** [comment]
No, FeatherPad doesn't support that yet. It's a good feature to add later.

BTW, when V0.8 is released. It'll have several new features. I wanted to release V0.8 in 2 weeks but with these new features, it may take a little longer.

**Comment by tsimonq2** [comment]
OK, is this something you can have done by Ubuntu 18.04's Feature Freeze (March 1st) or will I have to file a Feature Freeze Exception?

**Comment by tsujan** [comment]
I'll surely make V0.8 ready before the next Ubuntu LTS comes out.

**Comment by tsujan** [comment]
Oh, Qt is so cool! This is already supported and needs no change.

Qt (actually, `QSetting`) supports `$XDG_CONFIG_DIRS`. I tested with `/etc/featherpad/fp.conf`, removed `~/.config/featherpad/fp.conf` and the `etc` conf was loaded correctly.

On my system:
```bash
echo $XDG_CONFIG_DIRS
/etc:/etc/xdg:/usr/share
```
So, all of those directories could be used for a global config.

Please note that the conf file should be inside the "featherpad" folder like `featherpad/fp.conf`.

You could test by using Ubuntu's `$XDG_CONFIG_DIRS`, preferably with the latest git FeatherPad.

That's what distro's may want to do but there's no need for an `etc` config to be included in FeatherPad because FeatherPad already has its default settings.

**Comment by tsujan** [comment]
And please don't forget to remove the `[sessions]` section if you want to have a global conf. It'll do no harm if included but a clean global conf shouldn't have unneeded sections.

You could also show me your test global conf when ready.

**Comment by tsujan** [comment]
Of course, a setting like [text] → recentFiles isn't needed by a custom global conf either.

**Comment by agaida** [comment]
Bad luck - (l)ubuntu plays the XDG-Games other than all the other kids - so no /usr/share, not /etc - but /etc/xdg/$foo/ should be sufficient

**Comment by tsujan** [comment]
> Bad luck - (l)ubuntu plays the XDG-Games other than all the other kids

Strange! However, `$XDG_CONFIG_DIRS` can't be unset by any distro (at least, because of Qt) and even one place is enough -- in this case, `/etc/xdg` → `/etc/xdg/featherpad/fp.conf`.

**Comment by tsujan** [comment]
[Qt doc](http://doc.qt.io/qt-5/qstandardpaths.html) is kind to Ubuntu and gives "~/.config" and "/etc/xdg" as examples of `QStandardPaths::ConfigLocation` but I tested on my system and it was:
```
"~/.config", "/etc", "/etc/xdg", "/usr/share"
```
So, it's always the same as `$XDG_CONFIG_DIRS`.

**Comment by tsimonq2** [comment]
> And please don't forget to remove the [sessions] section if you want to have a global conf. It'll do no harm if included but a clean global conf shouldn't have unneeded sections.
>
> You could also show me your test global conf when ready.

Sure, thank you!

> Bad luck - (l)ubuntu plays the XDG-Games other than all the other kids - so no /usr/share, not /etc - but /etc/xdg/$foo/ should be sufficient

This is a bug and won't stay that way for long; I just have to hunt down the right people. :smile:

**Comment by tsujan** [comment]
Looking forward to your conf.

I'm doing the same thing with Kvantum, in case a distro wants to set a start theme other than Kvantum's default. In Kvantum, I intentionally didn't use this feature of `QSettings`.

**Comment by tsujan** [comment]
FYI:

I'll release the next version of FeatherPad (Kvantum too) a little after the "final freeze" of Ubuntu 18.04 -- which should be 19th April, as they promised -- because I want to install it on VirtualBox and make deb packages for the release pages as usual.

Until then, I'll hunt and fix small issues.

**Comment by tsimonq2** [comment]
> I'll release the next version of FeatherPad (Kvantum too) a little after the "final freeze" of Ubuntu 18.04 -- which should be 19th April, as they promised

Please be careful. If you delay it any more than that Sunday or Monday (at minimum) it won't make it into the release. Even waiting until after Final Freeze makes it a royal PITA.

So if you can avoid it, please don't wait that long (I can fund your beverage of choice!), but I guess do what you can.

Thanks.

**Comment by tsujan** [comment]
> So if you can avoid it, please don't wait that long

OK, I could easily release the next version of FeatherPad in 30th March, without attaching an Ubuntu deb package (which I don't think is needed anymore).

**Comment by tsimonq2** [comment]
> without attaching an Ubuntu deb package (which I don't think is needed anymore)

Right, at that point we're just duplicating work. :wink:

**Comment by tsujan** [comment]
All right! 30th March is more than enough for me :)

**Comment by tsujan** [comment]
Released 0.8 right now.

**Comment by tsimonq2** [comment]
> Released 0.8 right now.

Synced @agaida's upload to Ubuntu (technically breaking Feature Freeze, but who cares :P).
